### PR TITLE
Ft: add content guidelines page

### PR DESCRIPTION
**Pull request by advl**
## Done

- Added Content Guidelines page from [copydoc](https://docs.google.com/document/d/12nLPp3ktElJCNl_i3q5fGOjWMPhPXXFsIJGH-AuxduQ/edit?tab=t.0)

This is a draft. A few things are missing. Would love to hear your thoughts @bartaz @jmuzina on those : 
- Display the images in reasonable sizes
- Figure out a better way to display notessuch as in "contact form submission"

## QA

- Run vanilla-framework (`dotrun`)
- Go to Docs/Content Guidelines
- Review documentation

### Check if PR is ready for release

If this PR contains Vanilla SCSS or macro code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention
  - if existing APIs (CSS classes & macro APIs) are not changed it can be a bugfix release (x.x.**X**)
  - if existing APIs (CSS classes & macro APIs) are changed/added/removed it should be a minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)

## Screenshots

[if relevant, include a screenshot or screen capture]

### CODE REVIEW

File: templates/docs/content-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+- Aim for 3-6 words per line, no more than 3 lines per message.
+- Avoid asterisks (users don’t really trust them).
+
+### CTAs
+
+- Match the text of the primary CTA button to the action in the title.
+- Note: our current convention is to make ‘Cancel’ the secondary CTA button. It performs the same action as an ‘X’ icon in the top right corner.
+
+## Notification library
```
Comment (jmuzina) [inline templates/docs/content-guidelines.md:164]: These may look better as an unordered list with linebreaks between the bold "item title" and the paragraph beneath it. Though I'm not sure if design has something else in mind, was there a mockup created for this page? CC @lyubomir-popov

Comment (bartaz) [reply templates/docs/content-guidelines.md:164]: Lists can easily get messy and require a manual fine-tuning in HTML.

I think it's OK to leave it as is (with subheaders) for quick release, and we can finetune it later.

File: templates/docs/content-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+**Error: 404 for Vanilla**  
+404 Page not found  
+We can’t find the page you’re looking for. File a bug if you think this might be an error.
+
+<div class="u-fixed-width"><img src="https://assets.ubuntu.com/v1/98071afb-page-not-found.png" alt="Example of a 404 page from Vanilla"></div>
+
+**Confirmation: Contact form submission**  
+‘Thank you for contacting us. A member of our team will be in touch shortly.’
+
+> Note: Follow the guidance as described for consistency and accessibility across all content.
```
Comment (bartaz) [inline templates/docs/content-guidelines.md:190]: I think this (and possibly the "notification" examples above, could work best as notification component).

Somewhat like we do in other places in the docs:
https://github.com/canonical/vanilla-framework/blob/main/templates/docs/base/forms.md?plain=1#L161-L166

On site: https://vanillaframework.io/docs/base/forms#disabled

File: templates/docs/content-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+We should be specific about objects and actions and not have duplicate terms. Our messages typically include three attributes:
+
+- **Title**
+- **Description**
+- **CTA**
+
+**Example from Anbox:**
+
+<div class="u-fixed-width"><img src="https://assets.ubuntu.com/v1/e53e5133-reset-dialog.png" alt="Example of a reset dialog in Anbox with contextualized title, description and CTA."></div>
```
Comment (bartaz) [inline templates/docs/content-guidelines.md:140]: For images it would be good to add some arbitrary size to img tag for them not to spill too large? `width=720` or something?

We have images in documentation layout page (they are naturally 720px there: https://vanilla-framework-5425.demos.haus/docs/layouts/documentation)

File: templates/docs/content-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+Screen readers follow the HTML reading order to read and interpret content for users.
+
+When an image conveys important information, alt text is how users who don’t see the images consume that information. When a screen reader gets to an image, it will read the alt text to describe the content or purpose. Screen readers ignore images with empty alt attributes.
+
+For complex images like graphs or diagrams, consider putting a longer description somewhere else on the page. For example, in a nearby paragraph or through ARIA attributes.
+
+### When to write alt text
+
+- Always write alt text when the image adds meaning.
+- Use an empty alt attribute (`alt=""`) if the image doesn’t add meaning. For example, the currency icons below are decorative and do not add meaning. [Learn more about decorative images](https://www.w3.org/WAI/tutorials/images/decorative/).
```
Comment (jmuzina) [inline templates/docs/content-guidelines.md:37]: > the currency icons below are decorative and do not add meaning

"currency icons" are mentioned but not included in the page. I think we are missing something from the copydoc here.

File: templates/docs/content-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+- **Charts and diagrams:** If the chart is already explained in the text, use an empty alt attribute (`alt=""`). For example:
+
+<img src="https://assets.ubuntu.com/v1/0dd0ff70-empty-alt-attribute.png" alt="A screenshot of a page that describes the LTS support cycle of Ubuntu. A table summarizing the information is provided afterwards, with an empty alt-text." width="600">
+
+- **Graphs:** Describe the key trend or insight depicted in the graph so users understand the main data point. For example:
+
+- **Logos:** Just use the brand name (for example, ‘Microsoft Azure,’ not ‘Microsoft Azure logo’). Use empty state if the company name is used in the title or a link label.
+
+- **Icons and isometric images:** [Add examples for when you do need alt text and when you don’t].
```
Comment (jmuzina) [inline templates/docs/content-guidelines.md:59]: There is some template language from the copydoc left over here.

File: templates/docs/content-guidelines.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+---
+
+## Buttons and other CTAs
+
+On Canonical websites (‘Sites’), we have three CTA patterns: primary (green buttons), secondary (white buttons), and tertiary (linked text with a chevron).
+
```
Comment (jmuzina) [inline templates/docs/content-guidelines.md:68]: I wonder if we should add examples of the buttons below this?

```suggestion
On Canonical websites (‘Sites’), we have three CTA patterns: primary (green buttons), secondary (white buttons), and tertiary (linked text with a chevron).

<a href="#" class="p-button--positive">Positive</a>
<a href="#" class="p-button">Secondary</a>
<a href="#">Tertiary;&rsaquo;</a>
```

File: templates/docs/content-guidelines.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+- **Logos:** Just use the brand name (for example, ‘Microsoft Azure,’ not ‘Microsoft Azure logo’). Use empty state if the company name is used in the title or a link label.
+
+- **Icons and isometric images:** [Add examples for when you do need alt text and when you don’t].
+
+---
+
+## Buttons and other CTAs
+
+On Canonical websites (‘Sites’), we have three CTA patterns: primary (green buttons), secondary (white buttons), and tertiary (linked text with a chevron).
```
Comment (jmuzina) [inline templates/docs/content-guidelines.md:67]: > secondary (white buttons)

this isn't necessarily true, these buttons are [black on dark theme](https://vanillaframework.io/docs/examples/base/button?theme=dark). Not really a blocker for this PR, as it's a problem with the content itself, but figured I'd flag it anyway.

File: templates/docs/content-guidelines.md (hunk 4 above)
Comment (jmuzina) [reply templates/docs/content-guidelines.md:37]: @advl has added the image. resolving

File: side-navigation.yaml [hunk 8]
```diff
@@ -1,3 +1,4 @@
+---
```
Comment (jmuzina) [inline side-navigation.yaml:1]: Nit: do we need these new characters at the top of the file?

```suggestion
```

File: templates/docs/content-guidelines.md (hunk 5 above)
Comment (jmuzina) [reply templates/docs/content-guidelines.md:61]: The "icons and isometric images" are separate bullets in the copydoc, here you have combined them.

File: templates/docs/content-guidelines.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
+- **Card images:** Images inside cards or with links can be tricky. The screen reader will include the card title and link text, so alt text just needs to fill in any gaps. For example below, the card and link text tell the user a Dell XPS 13 Plus is a featured device, the missing information is it’s a laptop.
+
+<img src="https://assets.ubuntu.com/v1/1db0cd77-card-image-featured-device.png" alt="" width="250">
+
+- **Charts and diagrams:** If the chart is already explained in the text, use an empty alt attribute (`alt=""`). For example:
+
+<img src="https://assets.ubuntu.com/v1/0dd0ff70-empty-alt-attribute.png" alt="A screenshot of a page that describes the LTS support cycle of Ubuntu. A table summarizing the information is provided afterwards, with an empty alt-text." width="600">
+
+- **Graphs:** Describe the key trend or insight depicted in the graph so users understand the main data point. For example:
```
[the remark this answers is not in the document]
Comment (jmuzina) [reply templates/docs/content-guidelines.md:57]: We should at least remove the "for example:" at the end of the line so that reading the page does not feel strange - then flag that as a separate issue.

File: side-navigation.yaml (hunk 8 above)
Comment (jmuzina) [reply side-navigation.yaml:1]: @advl These characters are back, I wonder if you have some autoformatter for YAML that is adding these?

File: templates/docs/content-guidelines.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (jmuzina) [reply templates/docs/content-guidelines.md]: @advl Leaving to you to flag that as a separate issue, resolving the conversation so this can get merged
